src/mappings/entities/mapIdentities.ts

```diff
@@ -0,0 +1,192 @@
+import { Permissions } from '../../types/models/Permissions';
+import { Account } from '../../types/models/Account';
+import { Identity } from '../../types/models/Identity';
+import { getFirstValueFromJson, getTextValue } from '../util';
+
+/**
+ * Subscribes to the Identities related events
+ */
+export async function mapIdentities(
+  blockId: number,
+  eventId: EventIdEnum,
+  moduleId: ModuleIdEnum,
+  params: Codec[],
+  event: SubstrateEvent
+): Promise<void> {
+  const datetime = event.block.timestamp;
+  if (moduleId === ModuleIdEnum.Identity) {
+    if (eventId === EventIdEnum.DidCreated) {
+      const did = getTextValue(params[0]);
+      const address = getTextValue(params[1]);
+      await Identity.create({
+        id: did,
+        did,
+        primaryAccount: getTextValue(params[1]),
+        secondaryKeysFrozen: false,
+        blockId,
+        datetime,
+      }).save();
+
+      await Permissions.create({
+        id: address,
+        assets: null,
+        portfolios: null,
+        transactions: null,
+        transactionGroups: [],
+        createdBlockId: blockId,
+        datetime,
+      }).save();
+      await Account.create({
+        id: address,
+        identityId: did,
+        permissionsId: address,
+        eventId,
+        address,
+        blockId,
+        datetime,
+      }).save();
+    }
+
+    if (eventId === EventIdEnum.SecondaryKeysAdded) {
+      const { id: identityId } = await fetchIdentity(params[0]);
+
+      const accountsList = params[1].toJSON() as any[];
+
+      const promises = [];
+
+      for (const account of accountsList) {
+        const address = account.signer.account;
+
+        const { assets, portfolios, transactions, transactionGroups } = getPermissions(
+          account.permissions
+        );
+
+        promises.push(
+          Permissions.create({
+            id: address,
+            assets,
+            portfolios,
+            transactions,
+            transactionGroups,
+            createdBlockId: blockId,
+            datetime,
+          }).save()
+        );
+
+        promises.push(
+          Account.create({
+            id: address,
+            address,
+            identityId,
+            permissionsId: address,
+            eventId,
+            blockId,
+            datetime,
+          }).save()
+        );
+      }
+
+      await Promise.all(promises);
+    }
+
+    if (eventId === EventIdEnum.SecondaryKeysFrozen) {
+      const identity = await fetchIdentity(params[0]);
+      identity.secondaryKeysFrozen = true;
+      await identity.save();
+    }
+
+    if (eventId === EventIdEnum.SecondaryKeysUnfrozen) {
+      const identity = await fetchIdentity(params[0]);
+      identity.secondaryKeysFrozen = false;
+      await identity.save();
+    }
+
+    if (eventId === EventIdEnum.SecondaryKeysRemoved) {
+      await fetchIdentity(params[0]);
+      const accountsRemoved = params[1].toJSON() as any[];
```

Comment:
Is it possible to avoid any casts here? We should be able to cast to something more specific.

`any` basically opts out of TS checking, I think it should be avoided in the actual code. (alright occasionally in tests imo)

(few places this applies)